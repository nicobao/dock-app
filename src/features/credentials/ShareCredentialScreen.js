import React from 'react';
import {addTestId} from '../../core/automation-utils';
import {
  BackButton,
  Header,
  LoadingButton,
  NBox,
  ScreenContainer,
  Typography,
} from '../../design-system';
import {translate} from '../../locales';
import {ScrollView} from 'native-base';
import {useCredentials} from './credentials';
import {
  SELECT_CREDENTIALS,
  useCredentialPresentation,
} from './hooks/credentialPresentation';
import {useSingleDID} from '../didManagement/didHooks';
import {SelectCredentialsComponent} from './components/SelectCredentialsComponent';
import {SelectDIDComponent} from './components/SelectDIDComponent';
import {useDIDAuth} from '../didManagement/didAuthHooks';
export function ShareCredentialScreen({
  credentials,
  selectedCredentials,
  setSelectedCredentials,
  onNext,
  step,
  dids,
  onSelectDID,
  isFormValid,
}) {
  return (
    <ScreenContainer {...addTestId('ShareCredentialScreen')}>
      <Header>
        <BackButton {...addTestId('BackButton')} />
      </Header>
      <ScrollView mt={7} marginLeft={3} marginRight={3}>
        {step === SELECT_CREDENTIALS ? (
          <SelectCredentialsComponent
            selectedCredentials={selectedCredentials}
            credentials={credentials}
            setSelectedCredentials={setSelectedCredentials}
          />
        ) : (
          <SelectDIDComponent dids={dids} handleChange={onSelectDID} />
        )}
      </ScrollView>
      <NBox mb={7} mx={7}>
        <LoadingButton
          isDisabled={!isFormValid}
          {...addTestId('PresentCredentialBtn')}
          full
          onPress={async () => {
            await onNext(dids);
          }}
          size="sm">
          <Typography variant="credentialShareTitle" mb={2}>
            {translate('navigation.next')}
          </Typography>
        </LoadingButton>
      </NBox>
    </ScreenContainer>
  );
}

export function ShareCredentialScreenContainer({route}) {
  const {deepLinkUrl} = route.params || {};

  const {credentials} = useCredentials({onPickFile: () => {}});
  const {
    selectedCredentials,
    setSelectedCredentials,
    onNext,
    step,
    onSelectDID,
    isFormValid,
  } = useCredentialPresentation(deepLinkUrl);
  const {dids} = useDIDAuth();

  useSingleDID(dids, onSelectDID);

  return (
    <ShareCredentialScreen
      selectedCredentials={selectedCredentials}
      setSelectedCredentials={setSelectedCredentials}
      credentials={credentials}
      onNext={onNext}
      step={step}
      dids={dids}
      onSelectDID={onSelectDID}
      isFormValid={isFormValid}
    />
  );
}
