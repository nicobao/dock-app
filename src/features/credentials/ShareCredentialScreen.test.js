import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {ShareCredentialScreen} from './ShareCredentialScreen';

const mockStore = configureMockStore();

describe('CredentialsScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(
      <ShareCredentialScreen
        selectedCredentials={{}}
        setSelectedCredentials={jest.fn()}
        credentials={[]}
        onPresentCredentials={jest.fn()}
        onNext={jest.fn()}
        step={0}
        dids={[]}
        onSelectDID={jest.fn()}
        isFormValid={true}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
