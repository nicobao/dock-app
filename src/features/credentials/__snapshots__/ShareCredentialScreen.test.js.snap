// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CredentialsScreen should render correctly 1`] = `
<RNCSafeAreaView
  accessibilityLabel="ShareCredentialScreen"
  style={
    Object {
      "backgroundColor": "#141417",
      "flex": 1,
    }
  }
  testID="ShareCredentialScreen"
>
  <AppGlobalHeader />
  <Memo(ForwardRef(Box))
    flex={1}
  >
    <Memo(ForwardRef(Box))
      flex={1}
    >
      <Styled(View)>
        <BackButton
          accessibilityLabel="BackButton"
          testID="BackButton"
        />
      </Styled(View)>
      <ForwardRef
        marginLeft={3}
        marginRight={3}
        mt={7}
      >
        <SelectCredentialsComponent
          credentials={Array []}
          selectedCredentials={Object {}}
          setSelectedCredentials={[MockFunction]}
        />
      </ForwardRef>
      <Memo(ForwardRef(Box))
        mb={7}
        mx={7}
      >
        <LoadingButton
          accessibilityLabel="PresentCredentialBtn"
          full={true}
          isDisabled={false}
          onPress={[Function]}
          size="sm"
          testID="PresentCredentialBtn"
        >
          <Styled(Component)
            mb={2}
            variant="credentialShareTitle"
          >
            Next
          </Styled(Component)>
        </LoadingButton>
      </Memo(ForwardRef(Box))>
      <QRCodeModal
        description="Credential Presentation"
        visible={false}
      />
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</RNCSafeAreaView>
`;
